### PR TITLE
phpunit 6

### PR DESCRIPTION
**Pull request by kokspflanze**
added phpunit 6 support
drop php 5.5

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,22 +28,14 @@ env:
...
-      env:
-        - DEPS=latest
     - php: 5.6
       env:
         - DEPS=lowest
     - php: 5.6
       env:
         - DEPS=locked
+        - CS_CHECK=true
+        - DEPS=locked
```
Comment (Xerkus) [inline .travis.yml]: DEPS appears twice here

File: .travis.yml [hunk 2]
```diff
@@ -78,11 +72,12 @@ before_install:
...
 install:
+  - travis_retry composer install $COMPOSER_ARGS
+  - if [[ $TRAVIS_PHP_VERSION =~ ^5.6 ]]; then travis_retry composer update $COMPOSER_ARGS --with-dependencies $LEGACY_DEPS ; fi
   - if [[ $DEPS == 'latest' ]]; then travis_retry composer update $COMPOSER_ARGS ; fi
   - if [[ $DEPS == 'lowest' ]]; then travis_retry composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
-  - if [[ $TEST_COVERAGE == 'true' ]]; then composer require --dev $COMPOSER_ARGS satooshi/php-coveralls ; fi
-  - travis_retry composer install $COMPOSER_ARGS
-  - composer show --installed
+  - if [[ $TEST_COVERAGE == 'true' ]]; then travis_retry composer require --dev $COMPOSER_ARGS $COVERAGE_DEPS ; fi
+  - composer show
```
Comment (Xerkus) [inline .travis.yml]: This hack will make composer show versions again in travis build logs
```
- stty cols 120
- COLUMNS=120 composer show
```

File: test/BaseInputFilterTest.php [hunk 3]
```diff
@@ -840,7 +822,7 @@ protected function createInputInterfaceMock(
             ->willReturn($breakOnFailure)
         ;
         if (($isValid === false) || ($isValid === true)) {
-            $input->expects($this->once())
```
Comment (Xerkus) [inline test/BaseInputFilterTest.php]: why did you change test expectations?

File: test/CollectionInputFilterTest.php [hunk 4]
```diff
@@ -367,15 +365,15 @@ protected function createBaseInputFilterMock(
...
-        $inputFilter = $this->getMock(BaseInputFilter::class);
+        $inputFilter = $this->createMock(BaseInputFilter::class);
         $inputFilter->method('getRawValues')
             ->willReturn($getRawValues)
         ;
         $inputFilter->method('getValues')
             ->willReturn($getValues)
         ;
         if (($isValid === false) || ($isValid === true)) {
-            $inputFilter->expects($this->once())
```
Comment (Xerkus) [inline test/CollectionInputFilterTest.php]: why did you change test expectations?

File: .travis.yml (hunk 2 above)
Comment (kokspflanze) [inline .travis.yml]: looks good, also smth for other modules

File: test/BaseInputFilterTest.php (hunk 3 above)
Comment (kokspflanze) [inline test/BaseInputFilterTest.php]: because of https://github.com/zendframework/zend-inputfilter/blob/master/test/BaseInputFilterTest.php#L692-L693

both skipp sometimes the `isValid` method in https://github.com/zendframework/zend-inputfilter/blob/master/src/BaseInputFilter.php#L259

File: test/CollectionInputFilterTest.php (hunk 4 above)
Comment (kokspflanze) [inline test/CollectionInputFilterTest.php]: added better workflow with https://github.com/zendframework/zend-inputfilter/pull/132/commits/682546ed45f8acd2df9507c80ff8020c60fe273e

File: test/BaseInputFilterTest.php (hunk 3 above)
Comment (Xerkus) [inline test/BaseInputFilterTest.php]: I see.

You changed the collection input filter to revert `any()`, do you intend to do the same here?

Comment (kokspflanze) [inline test/BaseInputFilterTest.php]: i tested some parts, but nothing works well, looks for me bit to magic these parts of the test.
